### PR TITLE
Add unix shebang for executables generated by analysis tests

### PR DESCRIPTION
**Pull request by keddad**
See #18940 for more details. I've also wanted to update the tests, but it seems that this code doesn't have any test coverage (or I couldn't find it). To be fair, it is probably simple enough to not need it.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/analysis/test/AnalysisTestActionBuilder.java [hunk 1]
```diff
@@ -47,6 +47,8 @@ public static void writeAnalysisTestAction(
     StringBuilder sb = new StringBuilder();
     if (ruleContext.isExecutedOnWindows()) {
       sb.append("@echo off\n");
+    } else {
+      sb.append("#!/bin/sh");
```
[the remark this answers is not in the document]
Comment (keddad) [reply src/main/java/com/google/devtools/build/lib/analysis/test/AnalysisTestActionBuilder.java]: Fixed that, thanks

[the remark this answers is not in the document]
Comment (keddad) [reply src/main/java/com/google/devtools/build/lib/analysis/test/AnalysisTestActionBuilder.java]: Some CI pipelines still fail, but it doesn't look like it is related to the changes

[the remark this answers is not in the document]
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/analysis/test/AnalysisTestActionBuilder.java]: I agree, I'm re-running it now but I'm not concerned about it.

[the remark this answers is not in the document]
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/analysis/test/AnalysisTestActionBuilder.java]: Yes, it passed on a re-run. All good, waiting for merge. Thanks!

**Pull request closed without merging**